eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -28,61 +29,79 @@ parameters:
-            $RgName = "rg-${{ parameters.ServiceDirectory }}-$($BaseName)"
-            $ResourceGroupExpiration = (Get-Date).ToUniversalTime().AddHours(24).ToString("o")
-            Write-Host "Creating Resource Group: $($RgName)"
-            Write-Host "##vso[task.setvariable variable=AZURE_RESOURCEGROUP_NAME]$RgName"
+        - ${{ if eq(parameters['AllocateResourceGroup'], 'true') }}:
+          - pwsh: |
+              $BaseName = 't' + (New-Guid).ToString('n').Substring(0, 16)
+              $RgName = "rg-${{ parameters.ServiceDirectory }}-$($BaseName)"
+              $ResourceGroupExpiration = (Get-Date).ToUniversalTime().AddHours(24).ToString("o")
+              Write-Host "Creating Resource Group: $($RgName)"
+              Write-Host "##vso[task.setvariable variable=AZURE_RESOURCEGROUP_NAME]$RgName"
 
-            az login --service-principal -u $(provisioner-aad-id) -p $(provisioner-aad-secret) --tenant $(provisioner-aad-tenant)
-            az account set --subscription $(provisioner-subscription)
+              az login --service-principal -u $(provisioner-aad-id) -p $(provisioner-aad-secret) --tenant $(provisioner-aad-tenant)
+              az account set --subscription $(provisioner-subscription)
 
-            az group create --location westus2 --name $RgName --tags DeleteAfter=$ResourceGroupExpiration BuildId='$(Build.BuildId)' BuildJob='$(Agent.JobName)' BuildNumber='$(Build.BuildNumber)' BuildReason='$(Build.Reason)'
-            az group wait --created --resource-group $RgName
-          displayName: Create resource group and environment variable for tests
+              az group create --location westus2 --name $RgName --tags DeleteAfter=$ResourceGroupExpiration BuildId='$(Build.BuildId)' BuildJob='$(Agent.JobName)' BuildNumber='$(Build.BuildNumber)' BuildReason='$(Build.Reason)'
+              az group wait --created --resource-group $RgName
+            displayName: Create resource group and environment variable for tests
 
-      - template: ../steps/build-test.yml
-        parameters:
-          BuildTargetingString: ${{ parameters.BuildTargetingString }}
-          ServiceDirectory: ${{ parameters.ServiceDirectory }}
-          CoverageArg: $(CoverageArg)
-          EnvVars:
-            AZURE_RUN_MODE: 'Live' #Record, Playback
-            ${{ insert }}: ${{ parameters.EnvVars }}
-          BeforeTestSteps: ${{ parameters.BeforeTestSteps }}
-          AfterTestSteps: ${{ parameters.AfterTestSteps }}
-          PythonVersion: $(PythonVersion)
-          OSVmImage: $(OSVmImage)
-          ToxTestEnv: "whl"
-          AdditionalTestArgs: ${{ parameters.AdditionalTestArgs }}
-          TestMarkArgument: ${{ parameters.TestMarkArgument }}
-          InjectedPackages: ${{ parameters.InjectedPackages }}
-          BuildDocs: ${{parameters.BuildDocs}}
+          # figure out what's up with the create resource group above
+        - ${{ if eq(parameters['AllocateResourceGroup'], 'false') }}:
```

Comment:
We might want to add another parameter something like DeployArmTemplate or something like that as I don't think AllocateResourceGroup=false really tells folks what is happening. I think a better name will help folks understand what is happening. You can keep the AllocateResourceGroup for back-compat (assuming there is someone explicitly setting it), but in either case we probably should make the deployment model mutually exclusive so we don't end up deploying via both mechanisms.